sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyServiceClient.java

```diff
@@ -129,45 +129,45 @@ JsonWebKey transformSecretKey(SecretKey secretKey) throws JsonProcessingExceptio
         Objects.requireNonNull(encryptOptions, "'encryptOptions' cannot be null.");
 
         EncryptionAlgorithm algorithm = encryptOptions.getAlgorithm();
-        byte[] iv = encryptOptions.getIv();
-        byte[] authenticatedData = encryptOptions.getAdditionalAuthenticatedData();
         KeyOperationParameters parameters = new KeyOperationParameters()
             .setAlgorithm(algorithm)
             .setValue(encryptOptions.getPlainText())
-            .setIv(iv)
-            .setAdditionalAuthenticatedData(authenticatedData);
+            .setIv(encryptOptions.getIv())
+            .setAdditionalAuthenticatedData(encryptOptions.getAdditionalAuthenticatedData());
         context = context == null ? Context.NONE : context;
 
         return service.encrypt(vaultUrl, keyName, version, apiVersion, ACCEPT_LANGUAGE, parameters,
             CONTENT_TYPE_HEADER_VALUE, context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
             .doOnRequest(ignored -> logger.info("Encrypting content with algorithm - {}", algorithm.toString()))
-            .doOnSuccess(response -> logger.info("Retrieved encrypted content with algorithm- {}",
+            .doOnSuccess(response -> logger.info("Retrieved encrypted content with algorithm - {}",
                 algorithm.toString()))
             .doOnError(error -> logger.warning("Failed to encrypt content with algorithm - {}", algorithm.toString(),
                 error))
-            .flatMap(keyOperationResultResponse ->
-                Mono.just(new EncryptResult(keyOperationResultResponse.getValue().getResult(), algorithm, keyId)));
+            .flatMap(keyOperationResultResponse -> {
```

Comment:
Given that we are returning `Mono.just`, I would just change this to a `map` and return the created instance directly